lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -302,16 +312,41 @@ def write(chunk)
             entry['metadata']['severity'] = 'DEFAULT'
           end
 
+          if @service_name == CONTAINER_SERVICE
+            # TODO(a-robinson): Consider falling back to parsing the namespace,
+            # pod, and container names out of the log name if we don't have this
+            # information, which depends on the kubernetes_metadata_filter
+            # filter plugin.
+            labels = {}
+            field_to_label(record['kubernetes'], 'namespace_id', labels,
```

Comment:
Done, although it'll only be cleaner if the names coming out of the metadata plugin are the same as the names we want to send off to cloud logging, which'll depend on how you feel about them.